carbon-design-system/carbon-addons-iot-react
# fix(table): change tooltips to carbon-like

Closes #3510

**Summary**

- Change appearance of tooltips to carbon-like

**Change List (commits, features, bugs, etc)**

- Change `kind` prop in `TableToolbarSVGButton` to enable carbon tooltip in button
- Add i18n for icon description in `TableToolbarAdvancedFilterFlyout`
- Remove native browser tooltips
- Update test selector from `getByTitle` to `getById` or `getByText`
- Create `TableExpandRow` component with Carbon tooltip
- Create `IotOverflowMenu` component with Carbon tooltip
- Update storyshots

**Acceptance Test (how to verify the PR)**

- Go to [this story](https://deploy-preview-3541--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table--playground)
- Enable `toolbarAction`, `customToolbarContent`, `Demo download data as CSV` under "Toolbar" knob tab in order to see toolbar action buttons
- Change `hasFilter` value to `true` in "Sort & filter" knob tab to see filter action button
- Hover on the action buttons in table toolbar in order to see carbon tooltips.

</br>

- _Filter icon button_
- Go to [this story](https://deploy-preview-3541--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table--with-filtering)
- Change `hasFilter` value to `true` in knobs to see filter action button
- Hover on the action buttons in table toolbar in order to see carbon tooltips.

</br>

- _Flyout menu and data download icon buttons_
- Go to [this story](https://deploy-preview-3541--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table--with-toolbar)
- Enable `customToolbarContent`, `Demo download data as CSV` in knobs to see toolbar action buttons
- Hover on the action buttons in table toolbar in order to see carbon tooltips.

</br>

- _Row edit icon button_
- Go to [this story](https://deploy-preview-3541--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table--with-data-editing)
- Enable `hasRowEdit` in knobs to see edit button
- Hover on the button in table toolbar in order to see carbon tooltips.

</br>

- _Column selection icon button_
- Go to [this story](https://deploy-preview-3541--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table--with-asynchronous-data-source)
- Hover on the action buttons in table toolbar in order to see carbon tooltips.

</br>

- _Row expansion icon button_
- Go to [this story](https://deploy-preview-3541--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table--with-row-expansion)
- Hover on button to expand row.
- Verify that carbon tooltip has appeared.

</br>

- _Overflow menu icon button_
- Go to [this story](https://deploy-preview-3541--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table--with-aggregations)
- Hover on overflow menu.
- Verify that carbon tooltip has appeared.

</br>

- _TableCard_
- Go to [this story](https://deploy-preview-3541--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-dashboard--dashboard-all-cards-as-resizable)
- Enable tooltips in the story knobs.
- Scroll to the `TableCard`
- Hover on the toolbar.
- Verify that table icon-buttons have Carbon tooltip.

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here



**Things to look for during review**
